html_tag_collector/collector.py

```diff
@@ -240,12 +236,15 @@ def parse_response(url_response):
     Returns:
         list[dict]: List of dictionaries containing urls and relevant HTML tags.
     """
+    remove_whitespace = lambda s: " ".join(s.split()).strip()
```

Comment:
The name of this lambda function is a little misleading, as it technically removes  _extra_ whitespace. Recommend renaming, as it confused me initially.